engine/src/main/java/org/hibernate/search/engine/cfg/impl/PropertiesConfigurationPropertySource.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.cfg.impl;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+public class PropertiesConfigurationPropertySource extends MapConfigurationPropertySource {
+
+	public PropertiesConfigurationPropertySource(InputStream propertyStream) throws IOException {
+		super( buildProperties( propertyStream ) );
+	}
```

Comment:
If everything is going to be implemented in a static method whose result gets passed to the constructor, you don't really need a dedicated class. A static method in `MapConfigurationPropertySource` (e.g. `fromProperties`) would suffice.